tests/core/test_schema_components.py

```diff
@@ -179,6 +179,23 @@ def test_multi_index_index():
         schema.validate(df_fail)
 
 
+def test_single_index_multi_index_mismatch():
+    """Tests the failure case that attempting to validate a MultiIndex DataFrame
+    against a single index schema raises a SchemaError with a constructive error
+    message."""
+    ind = pd.MultiIndex.from_tuples(
+        [("a", "b"), ("c", "d"), ("e", "f")],
+        names=("one", "two"),
+    )
+    df_fail = pd.DataFrame(columns=["alpha", "beta"], index=ind)
```

Comment:
there's a weird bug in pandas 0.25.3 and python 3.8 that doesn't allow for empty dataframes:
```python
In [11]: pd.DataFrame(columns=[1,2,3])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-85419f40431c> in <module>
----> 1 pd.DataFrame(columns=[1,2,3])

~/miniconda3/envs/pandera-3.8/lib/python3.8/site-packages/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    409             )
    410         elif isinstance(data, dict):
--> 411             mgr = init_dict(data, index, columns, dtype=dtype)
    412         elif isinstance(data, ma.MaskedArray):
    413             import numpy.ma.mrecords as mrecords

~/miniconda3/envs/pandera-3.8/lib/python3.8/site-packages/pandas/core/internals/construction.py in init_dict(data, index, columns, dtype)
    240             else:
    241                 nan_dtype = dtype
--> 242             val = construct_1d_arraylike_from_scalar(np.nan, len(index), nan_dtype)
    243             arrays.loc[missing] = [val] * missing.sum()
    244

~/miniconda3/envs/pandera-3.8/lib/python3.8/site-packages/pandas/core/dtypes/cast.py in construct_1d_arraylike_from_scalar(value, length, dtype)
   1219     else:
   1220         if not isinstance(dtype, (np.dtype, type(np.dtype))):
-> 1221             dtype = dtype.dtype
   1222
   1223         if length and is_integer_dtype(dtype) and isna(value):

AttributeError: type object 'object' has no attribute 'dtype'
```

Only providing the `index` argument seems to work

```python
df_fail = pd.DataFrame(index=ind)
```

Reply:
e1799b26f8a7ae48b7705b9332e3453656943f9a